src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.Helpers.cs

```diff
@@ -44,44 +44,19 @@ internal static uint CountNumberOfLeadingAsciiBytesFromUInt32WithSomeNonAsciiDat
         {
             Debug.Assert(!AllBytesInUInt32AreAscii(value), "Caller shouldn't provide an all-ASCII value.");
 
-            // Use BMI1 directly rather than going through BitOperations. We only see a perf gain here
-            // if we're able to emit a real tzcnt instruction; the software fallback used by BitOperations
-            // is too slow for our purposes since we can provide our own faster, specialized software fallback.
-
-            if (Bmi1.IsSupported)
```

Comment:
I tried to preserve the intent of this one, which was that hardware accelerated `tzcnt` should be used on little-endian.  I can't think of any LE case that wouldn't be better off using `BitOperations` now, but I might have missed something.
cc @GrabYourPitchforks

Reply:
Thanks for the heads-up! :)
I'm not familiar enough with the larger change but this part LGTM.